pkg/driver/trace_driver.go

```diff
@@ -100,13 +103,220 @@ func (d *Driver) runCSVWriter(records chan interface{}, filename string, writerD
 	writerDone.Done()
 }
 
-func DAGCreation(functions []*common.Function) *list.List {
-	linkedList := list.New()
-	// Assigning nodes one after another
-	for _, function := range functions {
-		linkedList.PushBack(function)
+func createDAGWorkflow(functionList []*common.Function, function *common.Function, maxWidth int, maxDepth int) *list.List {
```

Comment:
Split this function into smaller functions with some meaningful actions. Then I can have a look again.

Reply:
Hi, I have updated the code based on your feedback. Do let me know if there are any issues. Thanks

Reply:
Hi @cvetkovic. May I ask if there are any issues / changes needed regarding the code? Thanks